### PR TITLE
Add option to toggle ruler visibility

### LINKED ISSUE #1416: Add a View > Rulers toggle to disable rulers
You will store the rulers enabled/disabled status next to `pub view_mode: ViewMode,` in `document_message_handler.rs`, as a boolean called `rulers_visible: bool,`.

Then you need to add a `DocumentMessage` message and its handler which enables/disables that, you can call it `SetRulersVisible { visible: bool }`.

Then add an entry in the view section of `menu_bar_message_handler.rs` for View > Rulers. It should display a checkbox to the left so take a look at how the other menu bar entries which do have a checkbox are implemented and use that approach to replicate how they tie the value to the `rulers_visible: bool,` field of the `DocumentMessageHandler`. Clicking that entry should call your new `SetRulersVisible` message.

You also need to add a FrontendMessage called `UpdateRulersVisibility { visible: bool }` and add a handler in the frontend for it in the correct place to hide the rulers from being rendered by the Document.rs svelte component.

### PR DESCRIPTION
**Pull request by zeroishero**
Closes #1416 
I added the menu, and made it send message to frontend to conditionally render the rulers element in svelte. I am using UpdateDocumentRuler and sending visible which is updated to rulersVisible. It works as intended but the markings and text is not being displayed. Here, as far as I know, the values necessary for computing markings and text is sent using UpdateDocumentRuler, and it has the data. I also tried updating spaces and interval and it did update it. 

I watched the contribution video as guided by @Keavon and the process does match. I had tried making ToggleRuler message but it didn't send the data related to rulers along with it.

### CODE REVIEW

**Comment by 0HyperCube** [comment]
@zeroishero the issue is that the svg for the numbers is set with width=0 and height=0. This can be fixed by adding `onMount(resize);` to `RulerInput.svelte`. You may need to add `import { onMount } from "svelte";` at the top of the js script.

**Comment by 0HyperCube** [comment]
Note that the tests failed to because of the new `rulers_visible` field. To fix this you can simply set serde to use a default value of true - see https://github.com/serde-rs/serde/issues/1030#issuecomment-522278006.

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+	"name": "Graphite",
+	"lockfileVersion": 3,
+	"requires": true,
+	"packages": {}
+}
```
Comment (0HyperCube) [inline package-lock.json:6]: This file can be removed (I presume it was accidentally added).

**Review by 0HyperCube — APPROVED** [review]
Works well, thanks for this contribution.

**Comment by Keavon** [comment]
I just completed my code review and pushed some improvements split out into several commits. So you can learn from the changes I made, I'd encourage you to read over each of the commits. One other note for future consideration: when submitting PRs, please do so from a branch in your fork other than `master` so we can check out your branch without its name conflicting with our copy of master, and to make it easier for you to keep your local fork up to date with our master branch (since there will be conflicting history when your work is based on master). Thanks for this feature, congrats on your contribution!

**Comment by Keavon** [comment]
!build

**Comment by zeroishero** [comment]
I was waiting for it to be reviewed and planned to make recommended changes myself. Anyway, thanks. I will look at changes.

**Comment by Keavon** [comment]
Yeah, sometimes we do code review by leaving comments, other times it's just easier to fix the things directly and push them. Having you fix the comments is better for your learning but more effort to describe on my part, so I took the more efficient route this time. I hope you can learn from the commit message + implementation to see. I also followed this up with one more commit which fixed a bug that I introduced (but didn't notice until after merging), so consider this change to be a part of the things you look at to learn through osmosis: 9224ed9ec6a6be8d378aa8508d50b34471216a73

Thanks again!

**Comment by Keavon** [comment]
Hi @zeroishero, thanks again for this code contribution to the project! We're still hoping you will respond to the request to relicense this code. Please see #4208 ASAP, thank you!

**Comment by Keavon** [comment]
Hi @zeroishero, we're grateful of your past code contribution to Graphite, but unless we hear from you by the **end of this weekend** with agreement about relicensing your code to include the MIT license, we will begin steps to remove your code contributions **so that you will no longer be a Graphite code contributor**. We'd much rather avoid this outcome, so please respond now.

To agree, simply log into your GitHub account, visit https://github.com/GraphiteEditor/Graphite/pull/4208, and copy-paste this into a comment:

"I license my past and future contributions to Graphite under the dual MIT/Apache-2.0 license, allowing licensees to choose either at their option."

If you're hesitant because you disagree or have questions, feel free to respond with another message; we'll do our best to answer your questions or concerns. More explanation is at the top of the linked thread if you're confused. **So far, 164 contributors have agreed and you are among the 11% remaining** whom we are still hoping to hear from before we begin removing your code from the project starting this Monday. (If you only see this later than Monday, your future belated response will still be appreciated.)

Thank you for your cooperation and support of open source, and we hope you'll help us keep your valued contributions in-tact.

**Pull request merged**